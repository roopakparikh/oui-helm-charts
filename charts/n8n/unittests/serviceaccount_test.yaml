suite: test service account

templates:
  - serviceaccount.yaml

tests:
  - it: should be empty when service account is not created
    set:
      serviceAccount.create: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should match snapshot of default values
    set:
      serviceAccount.annotations:
        my-key: my-value
    asserts:
      - equal:
          path: metadata.annotations.my-key
          value: my-value

  - it: should disable service account automaunt when automount flag set false
    set:
      serviceAccount.automount: false
    asserts:
      - equal:
          path: automountServiceAccountToken
          value: false

  - it: should match snapshot of default values
    set:
      serviceAccount.annotations:
        eks.amazonaws.com/role-arn: "arn:aws:iam::111122223333:role/iam-role-name"
    release:
      name: n8n
      namespace: n8n
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
