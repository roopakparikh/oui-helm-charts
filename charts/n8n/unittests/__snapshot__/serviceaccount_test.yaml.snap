should match snapshot of default values:
  1: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      annotations:
        eks.amazonaws.com/role-arn: arn:aws:iam::111122223333:role/iam-role-name
      labels:
        app.kubernetes.io/instance: n8n
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: n8n
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: n8n-1.0.0
      name: n8n
